test/lib/generateToken.test.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+const crypto = require('crypto');
+
+sinon.assert.expose(assert, { prefix: '' });
+require('sinon-as-promised');
+
+describe('generateToken', () => {
+    const randomBytes = 'some random bytes';
+    // Result of passing 'some random bytes' through a sha256 hash, in base64
+    const hash = 'mF0EvjvxWMrVz5ZGJcnbe0ZPooUlv/DAB9VrV6bmZmg=';
+    let nodeifyMock;
+    let generateToken;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+        nodeifyMock = {
+            withContext: sinon.stub()
+        };
+
+        mockery.registerMock('./nodeify', nodeifyMock);
+
+        // lazy load generateToken so it registers the mocks
+        /* eslint-disable global-require */
+        generateToken = require('../../lib/generateToken');
+        /* eslint-enable global-require */
+    });
+
+    beforeEach(() => {
+        nodeifyMock.withContext.resolves({
+            toString: () => randomBytes
+        });
+    });
+
+    after(() => {
+        mockery.deregisterAll();
+        mockery.disable();
+    });
+
+    it('generates a value', () =>
+        generateToken.generateValue()
+            .then((value) => {
+                assert.strictEqual(value, randomBytes);
+                assert.calledWith(nodeifyMock.withContext, crypto, 'randomBytes', [32]);
+            }));
+
+    it('hashes a value', () => {
+        assert.strictEqual(generateToken.hashValue(randomBytes), hash);
```

Comment:
Should probably verify that a different string does not produce the same hash. All this test says is that the function is returns the value of `hash`